### DISCUSSION TITLE
Retrieve Message Contents From AzureServiceBusReceiveMessageOperator

### DISCUSSION THREAD

**Discussion by JeffB-PPS** [apache/airflow#26446] [Q&A]
We're looking to leverage Airflow to receive messages from an Azure ServiceBus Queue and then pass them along to subsequent tasks.  The operator we're trying to use is AzureServiceBusReceiveMessageOperator.  One thing we can't figure out is how to get at the contents of the dequeued message after calling AzureServiceBusReceiveMessageOperator.  We see in the source code that the message is logged (and we do in fact see that in our logs), but unsure of how to get at it from within the DAG.

Appreciate any guidance someone can toss our way.

**Comment by potiuk** [comment]
You should likely write your custom operator/Hook. The operator does not seem to do anything more than just logging the messages. Likely you need to write/extend the Message Hook and return the messages returned because they seem to not be returned now. Even better you can likely modify existing hook and contribute it back - then you coud become one of the almost 2200 contributors to airflow and contribute back.

**Comment by perry2of5** [comment]
I've put in a PR to address this: 41601
